core/vm/contracts.go

```diff
@@ -231,6 +237,26 @@ func (c *sha256hash) Run(input []byte) ([]byte, error) {
 	return h[:], nil
 }
 
+type sum3 struct{}
+
```

Comment:
Is it worth splitting the specific pre-compiles into a separate file per precompile (thinking yes if there are going to be 10's of pre-compiles, no if less)

Reply:
This is `native stateless` precompile, they all are placed in one file here: https://github.com/Kava-Labs/go-ethereum/blob/master/core/vm/contracts.go
I just extended it with one more example stateless smart contract.

New stateful smart contracts is placed one per golang package.

fyi: PR was moved to https://github.com/Kava-Labs/go-ethereum/pull/10